internal/cluster/validator.go

```diff
@@ -498,31 +498,44 @@ func (v *clusterValidator) printIsDNSDomainDefined(context *clusterPreprocessCon
 	}
 }
 
-func (v *clusterValidator) noCidrsOverlapping(c *clusterPreprocessContext) ValidationStatus {
-	if swag.BoolValue(c.cluster.UserManagedNetworking) && !common.IsSingleNodeCluster(c.cluster) {
-		if !validationStatusToBool(v.isClusterCidrDefined(c)) || !validationStatusToBool(v.isServiceCidrDefined(c)) {
-			return ValidationPending
+func checkCidrsOverlapping(cluster *common.Cluster) error {
+	//Currently, the networks arrays can hold up to 2 subnets, one for each family
+	//in the same order. If machine networks is defined we assume it follows the
+	//same convension
+	var machineNetworkCidr, clusterNetworkCidr, serviceNetworkCidr string
+	for index := range cluster.ClusterNetworks {
+		if index < len(cluster.MachineNetworks) {
+			machineNetworkCidr = string(cluster.MachineNetworks[index].Cidr)
 		}
-	} else {
-		if !validationStatusToBool(v.isMachineCidrDefined(c)) || !validationStatusToBool(v.isClusterCidrDefined(c)) || !validationStatusToBool(v.isServiceCidrDefined(c)) {
-			return ValidationPending
+		clusterNetworkCidr = string(cluster.ClusterNetworks[index].Cidr)
+		serviceNetworkCidr = string(cluster.ServiceNetworks[index].Cidr)
+
+		if err := network.VerifyClusterCIDRsNotOverlap(machineNetworkCidr,
+			clusterNetworkCidr, serviceNetworkCidr,
+			network.IsMachineNetworkRequired(cluster)); err != nil {
+			return err
 		}
 	}
+	return nil
+}
+
+func (v *clusterValidator) noCidrsOverlapping(c *clusterPreprocessContext) ValidationStatus {
+	//If one of the required Cidr fields is empty return Pending status
+	if !validationStatusToBool(v.isClusterCidrDefined(c)) || !validationStatusToBool(v.isServiceCidrDefined(c)) {
+		return ValidationPending
+	}
+	if network.IsMachineNetworkRequired(c.cluster) && !validationStatusToBool(v.isMachineCidrDefined(c)) {
+		return ValidationPending
+	}
 
 	// TODO MGMT-7587: Support any number of subnets
 	// Assumes that the number of cluster networks equal to the number of service networks
-	for index := range c.cluster.ClusterNetworks {
-		machineNetworkCidr := ""
-		if len(c.cluster.MachineNetworks) > index {
-			machineNetworkCidr = string(c.cluster.MachineNetworks[index].Cidr)
-		}
+	if len(c.cluster.ClusterNetworks) != len(c.cluster.ServiceNetworks) {
```

Comment:
Not sure it needs to be checked here.  Maybe another validation ?

Reply:
I return validation error. This is just formulating the pre-assumption for this very validation in the code

Reply:
https://github.com/openshift/assisted-service/blob/d7580b33923b77f158e96eece5d65590ce16680e/internal/cluster/validator.go#L24